serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+
+## Hyperparameter Tuning
+
+If you’ve ever tried to tune hyperparameters for a machine learning model, you know that it can be a very painful process!
+
+To fine-tune your models you can check a variety of tools, check this article:
+
+- [A Comprehensive List of Hyperparameter Optimization & Tuning Solutions](https://medium.com/@mikkokotila/a-comprehensive-list-of-hyperparameter-optimization-tuning-solutions-88e067f19d9)
+
+In particular, if you are learning [Keras](https://www.tensorflow.org/guide/keras) as a top-level framework on the top of Tensorflow, Virgilio suggests you check [Hyperas](https://github.com/maxpumperla/hyperas).
+
+A very simple convenience wrapper around [Hyperopt](https://github.com/hyperopt/hyperopt) for fast prototyping with Keras models. Hyperas lets you use the power of hyperopt without having to learn the syntax of it. Instead, just define your Keras model as you are used to, but use a simple template notation to define hyper-parameter ranges to tune.
+
+## Serving Models
+
+After training Machine Learning models, you may want to make them available as a service, to your users, or to other computer systems that may use them.
+The most general way to serve a trained ML model is to use a library like [Flask](https://github.com/pallets/flask) API.
+
+Here are some resources to get started with this framework:
+
+- [Learn Flask for Python - Full Tutorial](https://www.youtube.com/watch?v=Z1RJmh_OqeA)
+- [The Mega Flask Tutorial](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-i-hello-world)
+
+Then follow these tutorials to learn how to use Flask to serve your trained models:
+
+- [A Flask API for serving scikit-learn models](https://towardsdatascience.com/a-flask-api-for-serving-scikit-learn-models-c8bcdaa41daa)
+- [Deploying Keras models using TensorFlow Serving and Flask](https://towardsdatascience.com/deploying-keras-models-using-tensorflow-serving-and-flask-508ba00f1037)
+
+The best way to understand something is _to do it_! So, check your knowledge with this 2-hours guided project from Coursera.
+
+- [Project: Deploy Models with TensorFlow Serving and Flask](https://www.coursera.org/learn/deploy-models-tensorflow-serving-flask)
+
+## Miscellaneous
+
+Virgilio lists here for you a bunch of incredible good GitHub pages with lists of hundreds of tools and libraries you can find useful when working in the Data Science environment. 
+What is the best way to use these pages? 
+The answer is that every problem has different characteristics and caveats, and you should be able to understand _if you need_ to learn a new tool, or you can get the things done with those that you already know! 
+So, use these pages to compare the pros and cons of the hundreds of tools that exist!
+
+- [Awesome AI-ML-DL](https://github.com/neomatrix369/awesome-ai-ml-dl)
```

Comment:
Thanks for mentioning my link. 

I would also mention these as sub-sections under the main link:
- [Things to know](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/things-to-know.md)
- [Data](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/data/README.md#data)
- [Cloud/DevOps/Infra Tools](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/cloud-devops-infra/README.md#tools)
- [Visualisation](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/details/visualisation.md#visualisation)
- [Tools, libraries and resources](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/details/java-jvm.md#tools--libraries-other-resources)

Reply:
I added some inside the  guide